### ISSUE TITLE
within-subject effect size calculations

### ISSUE THREAD

**Issue by mllewis** [langcog/metalab-archive#113]
Mike pointed out [this post](http://jakewestfall.org/blog/index.php/2016/03/25/five-different-cohens-d-statistics-for-within-subject-designs/) by Jake Westfall, about different ways of calculating d for within subject designs. The issue is that the magnitude of d is very different depending on whether you calculate the difference between conditions at the group level or at the subject level.

I looked a bit into this in the case of our effect size script and I think this is actually an important issue we need to figure out. For within-subject designs, we are calculating d (mostly) from either t, group means, or subject differences. Calculating d from t is the same as d by subject differences. But d from subject differences and d from group means are very different from each other according to the Westfall simulations (4:1), and so we should not be treating them as the same.

**Comment by christinabergmann** [comment]
I agree that we should think about this issue a bit. I once for InWordDB correlated for all cases where I had both d from two means (no mean differences available) versus d from t and apart from a few outliers in the extremes (iirc) they were very nicely correlated. 
For most other MAs we don't have systematically entered both, right? Should we do this to check? I think this would be the best way to compare our data to Westfall's simulations. 
Same goes for variances, actually, non?

**Comment by mcfrank** [comment]
Just an update that Molly and I think we can look at this by using
compute_ES to add info about how the computation was done and looking for
effects of computation type. This is not a perfect method but it will give
us preliminary evidence on whether there is heterogeneity in computation
type within MAs (which would be the scenario we're worried about).

Systematic entry of both d stats is a stronger method but also will take
more work and isn't possible in all cases...

**Comment by shotsuji** [comment]
Hi, 
I had answered this in an email but for it to not get lost in space for completeness:

To follow up on this: The strategy I've employed in the vowel paper, and that is currently implemented in the ES calculation script for Metalab, is the one proposed by Morris & Deshon (2002), thus using the correction factor proposed in formula (11) for difference score/paired t based values. Essentially what it does is to take into account that these values are correlated.

So we are not directly comparing these values in their raw form.
Of course there is no ideal method to do is, and what is problematic about it is also that we do not have all correlations (thus impute them for empty cases).

Sho
